scheduler/executor/go_cron.go

```diff
@@ -0,0 +1,149 @@
+package executor
+
+import (
+	"context"
+	"fmt"
+	"github.com/flyteorg/flyteadmin/scheduler/executor/interfaces"
+	"github.com/flyteorg/flyteadmin/scheduler/repositories/models"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/contextutils"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/robfig/cron"
+	"runtime/debug"
+	"runtime/pprof"
+	"time"
+)
+
+
+type goCronMetrics struct {
+	Scope                  promutils.Scope
+	JobFuncPanicCounter prometheus.Counter
+}
+
+// GoCron Struct implementing the GoCronWrapper which is used by the scheduler for adding and removing schedules
+// Each scheduled job accepts scheduled time parameter which helps to know what the actual invocation time and use
+// that to send an execution to admin
+type GoCron struct {
+	jobsMap map[string]interfaces.GoCronJobWrapper
```

Comment:
this needs to be thread-safe. this is a data-race?
The Register and Deregister is called from goroutines right?